### PR TITLE
GitHub Actions: cancel old but running workflows of a PR when pushing again

### LINKED ISSUE #2062: [ENH] GitHub Actions: cancel old but running workflows of a PR when pushing again
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

**Describe the solution you'd like**
E.g. use this:
https://github.com/marketplace/actions/cancel-workflow-action

### PR DESCRIPTION
**Pull request by RishiKumarRay**
#### Reference Issues/PRs
 Fixes #2062 

#### What does this implement/fix? Explain your changes.
GitHub Actions: cancel old but running workflows of a PR when pushing again

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Review by aiwalter — COMMENTED** [review]
Can younsee my draft PR #2061 ? I think for repos like sktime where we have PRs from forks we need to do sth as I did? How can we test your PR?

**Comment by RishiKumarRay** [comment]
@aiwalter thanks i have updated as yours, I didn't knew you had created this earlier, Thanks for giving this to me

**Comment by aiwalter** [comment]
Yeah I was not sure if my solution works, so nice that you look into as well 🙂

**Comment by RishiKumarRay** [comment]
@aiwalter i think both the solution will work, as  the other one i had already tried in a project but since you mentioned you want to have something like this for sktime so i think its good , One more thing i see you have a draft , I am happy to close this , Would you like to go for PR ?

**Review by aiwalter — DISMISSED** [review]

**Comment by lmmentel** [comment]
@RishiKumarRay Thanks a lot for putting in the time to look into this. As a test could you create a PR against your own fork and push some commits to check whenever the jobs are being cancelled properly?

**Comment by fkiraly** [comment]
@RishiKumarRay, kindly add yourself to `all-contributorsrc` with an `infra` contribution.

**Review by aiwalter — APPROVED** [review]

**Comment by aiwalter** [comment]
Pls fix linting

**Comment by RishiKumarRay** [comment]
@aiwalter can you please tell me what i have to fix since i am unable to find and fix?

**Comment by aiwalter** [comment]
You need to run pre-commit. Check our docs how to do if you need help

**Comment by RishiKumarRay** [comment]
@aiwalter can you please review now , I hope this works now

**Comment by aiwalter** [comment]
Pls make random changes like spaces in the files and commit again to run pre-commit

**Comment by RishiKumarRay** [comment]
@aiwalter Hi, i have done as you said and the the precommit has ran too
Screenshot from 2022-02-15 05-19-17
please check now

**Comment by aiwalter** [comment]
Obviously do it in all files you modifyed

**Comment by aiwalter** [comment]
The contributors file was not run locally

**Comment by RishiKumarRay** [comment]
@aiwalter Hi, thanks this time i did with contributors file too, please check now, I hope it works now

File: .all-contributorsrc [hunk 1]
```diff
@@ -1491,6 +1491,15 @@
         "doc",
         "infra"
       ]
+    },
+    {
+      "login": "RishiKumarRay",
+      "name":  "Rishi Kumar Ray",
```
Comment (aiwalter) [inline .all-contributorsrc:1497]:
```suggestion
      "name": "Rishi Kumar Ray",
```

File: .github/workflows/cancel.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Cancel Workflows on  Push
```
Comment (aiwalter) [inline .github/workflows/cancel.yml:1]:
```suggestion
name: Cancel Workflows on Push
```

**Review by aiwalter — APPROVED** [review]

**Comment by aiwalter** [comment]
@lmmentel pls check

**Comment by aiwalter** [comment]
Actually ,could we also cancel the other checks of the current run as soon as one check is failing?

**Comment by RishiKumarRay** [comment]
@aiwalter i don't think there is any direct action for this as far as i know, maybe i am wrong, also if we want to make this then we can try need field in the workflow which might make other jobs dependent and if one cancels other will be cancelled, let me know what do you think.

**Comment by aiwalter** [comment]
you could make a separate PR if you want for that. Lets first try to get this done :)

**Comment by RishiKumarRay** [comment]
@aiwalter I think the other two files i.e the test dir that are pushed here are the reason of failure as i read the logs and it says the files are already present. I don't have much idea about this file since it came after i installed pre commit and i see that these two files are already present so it would be better i should remove the files and only push contributorssrc and cancel.yml
what do you think?

**Comment by RishiKumarRay** [comment]
@aiwalter see this 
Screenshot from 2022-02-15 14-07-37

**Comment by aiwalter** [comment]
yes pls delete this, I didnt see that you added this dir before

**Comment by lmmentel** [comment]
@RishiKumarRay it seems you accidentally committed `testdir/.coveragerc` and `testdir/setup.cfg` please remove those files as they are already at the project root.

**Comment by lmmentel** [comment]
> Actually ,could we also cancel the other checks of the current run as soon as one check is failing?

I thought this action would cancel a **run** instead of a single **check** :thinking:

**Comment by RishiKumarRay** [comment]
@aiwalter @lmmentel i have deleted , please trigger the workflow now, I hope it works now

**Review by lmmentel — APPROVED** [review]
Looks good :+1:

**Comment by aiwalter** [comment]
@lmmentel yes I was thinking about canceling all jobs as soon as one is failing. This is currently not the case. E.g. code-quality jobs fails but all other tests are still running, which could be avoided. @RishiKumarRay would you like to work on that?

**Comment by lmmentel** [comment]
> @lmmentel yes I was thinking about canceling all jobs as soon as one is failing. This is currently not the case. E.g. code-quality jobs fails but all other tests are still running, which could be avoided. @RishiKumarRay would you like to work on that?

I'm not sure if that's the case since "code quality" is an explicit dependency for all other steps so they **should** be cancelled when "code quality" fails. Let me know if you find a case where it happens so that we can look into it.

**Comment by RishiKumarRay** [comment]
@aiwalter i would be happy to do this but as of now i am stuck with some other tasks so I won't be able to do but once i get free i will be able to do this. Thanks

**Pull request merged**